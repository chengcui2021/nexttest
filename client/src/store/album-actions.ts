import albumSlice from './album-slice'
import { AnyAction } from '@reduxjs/toolkit'
import { ThunkAction } from '@reduxjs/toolkit'
import { RootState } from './index'
import { AlbumModel } from "../models/redux-models";
import AlbumService from "../service/albumService";

export const albumActions = albumSlice.actions

export const fetchAlbums = (search: string): ThunkAction<void, RootState, unknown, AnyAction> => {
    return async (dispatch, getState) => {
        const response: AlbumModel[] = await AlbumService.getAllAlbums(search);
        dispatch(albumActions.setAlbums(response))
    }

}