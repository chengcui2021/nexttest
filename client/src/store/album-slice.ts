import { AlbumModel, AlbumArrayModel } from "../models/redux-models";
import { createSlice, PayloadAction } from "@reduxjs/toolkit";

const initialAlbumState: AlbumArrayModel = {
    all_albums: [],
}

const albumSlice = createSlice({
    name: 'album',
    initialState: initialAlbumState,
    reducers: {
        setAlbums(state, action: PayloadAction<AlbumModel[]>) {
            state.all_albums = action.payload;
        },
    }
})
export default albumSlice;