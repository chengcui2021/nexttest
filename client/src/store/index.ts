import {configureStore} from '@reduxjs/toolkit';
import albumSlice from './album-slice'

const store=configureStore(
    {
        reducer:{album:albumSlice.reducer}
    }
)

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch =typeof store.dispatch
export default store;