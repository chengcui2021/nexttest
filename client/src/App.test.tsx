import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { fetchAlbums } from './store/album-actions';

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares);
const store = mockStore({
  album: {
    all_albums: [
      { 
        artistName: 'Jack Johnson',
        collectionName: 'On and On',
        trackName: 'Taylor'
      }
    ]
  }
})

describe('test actions', () => {
  beforeEach(() => {
    store.clearActions();
  });
  describe("fetch albums with search term 'What Room Was the Holiday In'", () => {
    test('Dispatches the correct action and payload', () => {
      fetchAlbums('What Room Was the Holiday In');
      expect(store.getActions()).toMatchSnapshot();
    });
  });

});

