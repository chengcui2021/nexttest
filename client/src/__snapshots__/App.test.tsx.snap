// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gator_actions eatFood Dispatches the correct action and payload 1`] = `Array []`;

exports[`test actions fetch albums with search term 'What Room Was the Holiday In' Dispatches the correct action and payload 1`] = `Array []`;
